admin/services/data-access/check-window.data.service.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const CheckWindow = require('../../models/check-window')
+
+const checkWindowDataService = {
+
+  /**
+   * Fetch check window document by id.
+   * @param id
+   * @returns {Promise.<*>}
+   */
+  fetchCheckWindow: async (id) => {
```

Comment:
This is a specialised case of findOne - a generic findOne implementation would be more useful.  Also `.exec()` is missing.

```
async function findOne (criteria) {
   const checkWindow = await CheckWindow.findOne(criteria).exec()
   return checkWindow
}
```

```
myCheckWindow = checkWindowDataService.findOne({id: 1234})
```

Reply:
Generic? You mean directly written in the controller? I would still prefer to have all nicely grouped in services.

Reply:
@jon this is now a special case for check windows only (see changes).